File: Source/Platforms/Android/java/jp/co/virgintech/Data_io.java

package jp.co.virgintech.virgintech5stproject;

import android.app.Activity;
import android.content.Context;
//import android.content.res.Resources;
import android.preference.PreferenceManager;
import android.content.SharedPreferences;
import android.content.SharedPreferences.Editor;

public class Data_io
{
    static public void initialize_Preferences()
    {
    }
    static public void save_Coin_Value(Context context,int value)
    {
        SharedPreferences pref = PreferenceManager.getDefaultSharedPreferences(context);
        Editor e = pref.edit();
        e.putInt("coin", value);
        e.commit();
    }
    static public int load_Coin_Value(Context context)
    {
        int value;
        SharedPreferences pref = PreferenceManager.getDefaultSharedPreferences(context);
        value = pref.getInt("coin", 0);
        return value;
    }
    static public void save_Int_Value(Context context,String file,String key,int value)
    {
        SharedPreferences pref = context.getSharedPreferences(file, context.MODE_PRIVATE);
        Editor e = pref.edit();
        e.putInt(key, value);
        e.commit();
    }
    static public int load_Int_Value(Context context,String file,String key)
    {
        int value;
        SharedPreferences pref = context.getSharedPreferences(file, context.MODE_PRIVATE);
        value = pref.getInt(key, 0);
        return value;
    }
    static public void getResText(Context context,String type,String key)
    {
        /* Use Resources-Class
         Resources res = context.getResources();
        int resId = res.getIdentifier(key, "string", context.getPackageName());
        String str = res.getString(resId);
         */
        
        //Use Context-Class
        int resId = context.getResources().getIdentifier(key, type, context.getPackageName());
        String str = context.getString(resId);
        
        //CallBack Return!
        callbackStrings(str);
    }
    
    static private native void callbackStrings(String message);
}

File: Data_io.m
//
//  Data_io.m
//  VirginTech5stProject
//
//  Created by VirginTech LLC. on 2015/07/05.
//  Copyright (c) 2015年 Apportable. All rights reserved.
//

#import "Data_io.h"

@implementation Data_io

NSString* str;

//======================
// Objcからアクセス用
//======================
+(NSString*)getCallBackStrings{
    return str;
}

//======================
// Javaブリッジング
//======================

//コインセーブ・ロード（デフォルトプリファレンス）
@bridge (method, static)save_Coin_Value:value:  =save_Coin_Value;
@bridge (method, static)load_Coin_Value:        =load_Coin_Value;

//汎用タイプint型（プリファレンスファイル指定）
@bridge (method, static)save_Int_Value:file:key:value: =save_Int_Value;
@bridge (method, static)load_Int_Value:file:key:       =load_Int_Value;

//リソーステキスト取得（＊コールバックで戻す）
@bridge (method, static)getResText:type:key:      =getResText;
@bridge (callback, static) callbackStrings: = callbackStrings;
+(void)callbackStrings:(NSString *)message{
    str=message;
}

@end
